internal/types/codeintel.go

```diff
@@ -31,26 +31,29 @@ type CodeIntelAggregatedInvestigationEvent struct {
 type NewCodeIntelUsageStatistics struct {
-	StartOfWeek                                      time.Time
-	WAUs                                             *int32
-	PreciseWAUs                                      *int32
-	SearchBasedWAUs                                  *int32
-	CrossRepositoryWAUs                              *int32
-	PreciseCrossRepositoryWAUs                       *int32
-	SearchBasedCrossRepositoryWAUs                   *int32
-	EventSummaries                                   []CodeIntelEventSummary
-	NumRepositories                                  *int32
-	NumRepositoriesWithUploadRecords                 *int32
-	NumRepositoriesWithoutUploadRecords              *int32 // Deprecated, no longer sent
-	NumRepositoriesWithFreshUploadRecords            *int32
-	NumRepositoriesWithIndexRecords                  *int32
-	NumRepositoriesWithFreshIndexRecords             *int32
-	NumRepositoriesWithAutoIndexConfigurationRecords *int32
-	CountsByLanguage                                 map[string]CodeIntelRepositoryCountsByLanguage
-	SettingsPageViewCount                            *int32
-	UsersWithRefPanelRedesignEnabled                 *int32
-	LanguageRequests                                 []LanguageRequest
-	InvestigationEvents                              []CodeIntelInvestigationEvent
+	StartOfWeek                                               time.Time
+	WAUs                                                      *int32
+	PreciseWAUs                                               *int32
+	SearchBasedWAUs                                           *int32
+	CrossRepositoryWAUs                                       *int32
+	PreciseCrossRepositoryWAUs                                *int32
+	SearchBasedCrossRepositoryWAUs                            *int32
+	EventSummaries                                            []CodeIntelEventSummary
+	NumRepositories                                           *int32
+	NumRepositoriesWithUploadRecords                          *int32
+	NumRepositoriesWithoutUploadRecords                       *int32 // Deprecated, no longer sent
+	NumRepositoriesWithFreshUploadRecords                     *int32
+	NumRepositoriesWithIndexRecords                           *int32
+	NumRepositoriesWithFreshIndexRecords                      *int32
+	NumRepositoriesWithAutoIndexConfigurationRecords          *int32
+	CountsByLanguage                                          map[string]CodeIntelRepositoryCountsByLanguage
+	SettingsPageViewCount                                     *int32
+	UsersWithRefPanelRedesignEnabled                          *int32
+	LanguageRequests                                          []LanguageRequest
+	InvestigationEvents                                       []CodeIntelInvestigationEvent
+	CommitDistanceMean, CommitDistanceStddev                  *float64
+	CommitDistanceCount, CommitDistanceMax, CommitDistanceMin *int32
+	CommitDistanceP10, CommitDistanceP75, CommitDistanceP90   *int32
```

Comment:
Always per-week?

Reply:
Do we do anything else anywhere?

Reply:
No just want to make sure I know the span of events. Some places we care about Daily/Weekly/Monthly (like active users) but having weekly is fine. Might be worth it to name the wrapping field explicitly here.

Reply:
This type has a [StartOfWeek](https://github.com/sourcegraph/sourcegraph/pull/56195/files/5d87748b99597df69fe850f9e023a6b0fed47fe2..28848b1b7d7c9b7862188b655dd3e12f106f495d#diff-e105f1d7387a964b783c946bf88bf1850a7169c0b29243826d87c4fa08caf533R34) field that I assumed meant it was for weekly events

Reply:
Yeah but some things are not weekly counts, but just "at the moment" counts. Could do this to be more explicit:


```suggestion
	WeeklyAggregateCommitDistance *CodeIntelAggregatedCommitDistance`
```

Reply:
Updated it to be like EventSummaries, lmk wyt